include/pybind11/pybind11.h

```diff
@@ -452,6 +452,10 @@ class cpp_function : public function {
 
     /// When a cpp_function is GCed, release any memory allocated by pybind11
     static void destruct(detail::function_record *rec) {
+        #if !defined(PYPY_VERSION) && PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION == 9
```

Comment:
We could even move this to the location it's used, or not?

Reply:
That's where it was originally, but thought it was better not in the loop. Defining a static variable inside an if seems rather odd, but if the compiler handles it correctly, it's probably fine?

Reply:
I don't really mind, but yeah, that was my thinking

Reply:
The assembly when including a static variable definition is a bit more complex, using a guard variable and a lock, so I think it's better to just do it once, outside the loop. But I'm happy either way. https://gcc.godbolt.org/z/cMcsW9

Reply:
Right. Just keep it like this then, indeed! :-)